Add InverseGamma moment

morganstrom opened the pull request:
Simplifies calculation using aesara.tensor.switch statement

**Thank your for opening a PR!**

Depending on what your PR does, here are a few things you might want to address in the description:
+ [ ] what are the (breaking) changes that this PR makes?
+ [ ] important background, or details about the implementation
+ [ ] are the changes—especially new features—covered by tests and docstrings?
+ [ ] [linting/style checks have been run](https://github.com/pymc-devs/pymc3/wiki/PyMC3-Python-Code-Style)
+ [ ] [consider adding/updating relevant example notebooks](https://github.com/pymc-devs/pymc-examples)
+ [ ] right before it's ready to merge, mention the PR in the RELEASE-NOTES.md

morganstrom committed:
naive implementation without broadcasting, tests fail

zaxtax committed:
Adding Deterministic for sampling_numpyro

zaxtax committed:
Obey precommit hook

morganstrom committed:
simplifies logic of InverseGamma moment using switch statement and adds more test cases

morganstrom committed:
adds test case where one element in the output array is expected to be infinite

ricardoV94 commented on pymc/distributions/continuous.py:
```diff
@@ -2326,23 +2326,17 @@ def dist(cls, alpha=None, beta=None, mu=None, sigma=None, sd=None, *args, **kwar
+        m = beta / (alpha - 1.0)
+        if not rv_size_is_none(size):
+            m = at.full(size, m)
+        return at.switch(alpha > 1, m, np.inf)
```
We should either use the mode as the moment or switch between the mode and the mean, so that we don't have to return an infinite value. The goal of `get_moment` is to obtain a value that is always okay as a starting point for that distribution.

morganstrom replied:
Okay, I think I fixed it now! Let me know if I need to make further changes

ricardoV94 committed:
Fix `rvs_to_value_vars` inplace update bug

ricardoV94 committed:
Remove logcdfpt and simplify logp and logcdf helpers

These methods will only retrieve the RV logp and logcdf terms, and no longer attempt to replace any potential RVs in their arguments.
A ValueError message is raised when the provided value is not compatible with the original RV dimensions

zaxtax committed:
Extract values efficiently

zaxtax committed:
Change Normal to Uniform to test combinations of transforms and deterministic

zaxtax committed:
lint fix

Commit:
Adding moments for AsymmetricLaplace and SkewNormal and corresponding tests

zaxtax committed:
Use value_vars directly

Commit:
Adding moment for Rice distribution and associated test

yadav-sachin committed:
Add MvNormal moment (#5171)

 
Co-authored-by: Ricardo Vieira <28983449+ricardoV94@users.noreply.github.com>

morganstrom committed:
returns mode where mean is not defined

Commit:
Adding PolyaGamma moment and corresponding tests

ricardoV94 committed:
Temporarily pin aeppl to 0.0.15

ricardoV94 commented on pymc/tests/test_distributions_moments.py:
```diff
@@ -385,6 +386,26 @@ def test_gamma_moment(alpha, beta, size, expected):
+        (np.arange(1, 6), 1, None, np.array([0.5, 1, 1 / 2, 1 / 3, 1 / 4])),
+    ],
+)
+def test_inverse_gamma_moment(alpha, beta, size, expected):
```
Can we reduce a bit the number of tests?

morganstrom replied:
Better now?

aloctavodia committed:
BART: improve accuracy and other minor fixes (#5177)

* improve accuracy and other minor fixes

* update release notes

* fix typo

lucianopaz committed:
Add DensityDist moment (#5159)

* Add DensityDist moment

* Specialize get_moment for multivariate density dists

morganstrom committed:
removed redundant test cases

morganstrom commented:
I did `git fetch upstream` and `git rebase upstream/main`, as recommended here: https://github.com/pymc-devs/pymc/blob/35f9966f15e76d8ba304914be69e4115fdcfd4ef/CONTRIBUTING.md 

Is that why there are so many commits in this pull request suddenly? 

Don't know how to solve it easily, should I just create a brand new branch and add my changes there perhaps?
